helix-core/src/main/java/org/apache/helix/cloud/azure/AzureInstanceMetadataFetcher.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.helix.cloud.azure;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.common.cloud.CloudInformationFetcher;
+
+public class AzureInstanceMetadataFetcher implements CloudInformationFetcher {
+  private final String _azureUrl;
+
+  public AzureInstanceMetadataFetcher(String azureInstanceMetadataEndpoint) {
+    _azureUrl = azureInstanceMetadataEndpoint;
+  }
+
+  /**
+   * Get the instance metadata information from Azure using the azure instance metadata endpoint
```

Comment:
Let's make it clear that it's the VM instance metadata info?

Reply:
Not only for VM. For the ones using Helix TF may have the capability using topology information for task assignment. And most of instance type of TF instances are containers.